### ISSUE TITLE
Emitting "testPass" after each step seems wrong.

### ISSUE THREAD

**Issue by phenomnomnominal** [protractor-cucumber-framework/protractor-cucumber-framework#103]
Hi there, I have an issue that I'd like to discuss, and happy to have a crack at a PR if it makes sense!

It seems strange to me to emit the "testPass" event after any step completes. Shouldn't the "test" be the whole scenario?

It would be reasonable for any failing step to be able to fail the test, but I feel like it should only pass once the whole scenario is finished.

I am currently trying to write a protractor plugin that needs to step up some state before each scenario and tear it down after. I can only seems to do this per feature at the moment.

Thoughts?

**Comment by wswebcreation** [comment]
Hi @phenomnomnominal 

Just a question. If you have a scenario like:

```shell
Scenario: Edit user as an admin
Given I logon as a administrator
When I go to the admin panel
Then I'm able to edit user John Doe
```

And your `When I go to the admin panel` fails, what should be the benefit of running the `Then I'm able to edit user John Doe` if you are not able to open the edit page?

**Comment by phenomnomnominal** [comment]
> It would be reasonable for any failing step to be able to fail the test

I agree with you on that 😄  (sorry my original post wasn't so clear)

What I don't agree is that if I have in `protractor.conf.js`

```
// ...
    plugins: [{
        inline: {
            postTest () {
                console.log('post test');
            }
        }
    }]
```

Then `'post test'` will log out after each step that passes (e.g. after the `Given`, not each `Scenario`). 

This seems to be because of [this code](https://github.com/protractor-cucumber-framework/protractor-cucumber-framework/blob/master/lib/resultsCapturer.js#L190-L250) that runs after each step, and emits the 'testPass' event.

**Comment by wswebcreation** [comment]
Aha ;-),

But isn't this the downside of using a second framework to support Protractor? You are now referring to the lifecycle of Protractor / protractor-plugin to be "equal" to the lifecycle of CucumberJS (with protractor-cucumber-framework as an intermediate)? Or do I understand you wrong?

**Comment by phenomnomnominal** [comment]
Perhaps, but `testPass` is a fairly general part of any test lifecycle. Would you say that the test has passed after the first `Given` step? I personally wouldn't! 

Admittedly I haven't looked into the repercussions of this change... do you think they would be significant?

**Comment by wswebcreation** [comment]
Maybe `testPass` isn't a _good_ word, but yeah, a part of the test has passed.

I do think there is a reason to use `testPass` in this way but maybe @darrinholst explain / confirm this. 

By the way, I do like the discussing you started 👍, just to get more clarity ;-)

**Comment by darrinholst** [comment]
We used to emit it after each scenario. That changed [here](https://github.com/protractor-cucumber-framework/protractor-cucumber-framework/issues/63). I don't personally use an IDE, but I know a lot (relative to me ;) of people do so that would be taking away some niceties from them. Feels like Protractor needs some more granular events that tools could hook into instead of overloading testPass/testFail.

**Comment by phenomnomnominal** [comment]
Hmm yeah I suspected that might be the reason. 

I'm not even sure what extra event you could add that would make sense? testProgress or something? But that only makes sense in a cucumber-ish world. I'm going to dig around in protractor code some more and see what I find.

 I guess for now my plugin will have to detect that cucumber is being used and use cucumber hooks in that case.

**Comment by darrinholst** [comment]
You could add something [here](https://github.com/protractor-cucumber-framework/protractor-cucumber-framework/blob/master/lib/resultsCapturer.js#L236)...maybe a `progress` percentage or something. Your plugin should get the event details, right?

**Comment by darrinholst** [comment]
Closing in favor of #105. Let us know if that won't work for you.

**Comment by phenomnomnominal** [comment]
Yeah that's fine for me. I actually found a way around this, but I still think it's a good idea. I will also probably raise an issue on protractor about this stuff.
